s3torchconnector/tst/e2e/conftest.py

```diff
@@ -18,41 +18,68 @@ def getenv(var: str, optional: bool = False) -> str:
     return v
 
 
-class BucketPrefixFixture(object):
+class BucketPrefixData(object):
     """An S3 bucket/prefix and its contents for use in a single unit test. The prefix will be unique
     to this instance, so other concurrent tests won't affect its state."""
 
     region: str
     bucket: str
     prefix: str
     storage_class: str = None
+    contents: dict
 
     def __init__(
-        self, region: str, bucket: str, prefix: str, storage_class: str = None
+        self,
+        region: str,
+        bucket: str,
+        prefix: str,
+        storage_class: str = None,
+        contents: dict = None,
     ):
         self.bucket = bucket
         self.prefix = prefix
         self.region = region
         self.storage_class = storage_class
-        self.contents = {}
-        session = boto3.Session(region_name=region)
-        self.s3 = session.client("s3")
+        self.contents = contents or {}
 
     @property
     def s3_uri(self):
         return f"s3://{self.bucket}/{self.prefix}"
 
+    def __getitem__(self, index):
+        return self.contents[index]
+
+    def __iter__(self):
+        return iter(self.contents)
+
+
+class BucketPrefixFixture(BucketPrefixData):
+    """An S3 bucket/prefix and its contents for use in a single unit test. The prefix will be unique
+    to this instance, so other concurrent tests won't affect its state."""
+
+    def __init__(
+        self, region: str, bucket: str, prefix: str, storage_class: str = None
+    ):
+        super().__init__(region, bucket, prefix, storage_class)
+        session = boto3.Session(region_name=region)
+        self.s3 = session.client("s3")
+
     def add(self, key: str, contents: bytes, **kwargs):
         """Upload an S3 object to this prefix of the bucket."""
         full_key = f"{self.prefix}{key}"
         self.s3.put_object(Bucket=self.bucket, Key=full_key, Body=contents, **kwargs)
         self.contents[full_key] = contents
 
-    def __getitem__(self, index):
-        return self.contents[index]
+    def get_data_snapshot(self):
+        """
+        Returns a read-only copy of the current instance's data.
```

Comment:
nit: should comments start at the same line with """ or at the next line? Methods above/below are inconsistent.